### DISCUSSION TITLE
Splink 4.0.0: AttributeError: 'Linker' object has no attribute 'query_sql'

### DISCUSSION THREAD

**Discussion by aalexandersson** [moj-analytical-services/splink#2320] [Q&A] [answered]
I am trying to use `linker.query_sql()` which should still be used in Splink 4 according to this web page:
[https://moj-analytical-services.github.io/splink/topic_guides/splink_fundamentals/querying_splink_results.html](https://moj-analytical-services.github.io/splink/topic_guides/splink_fundamentals/querying_splink_results.html)

On my test data, this code worked in Splink 3.9.15 but it no longer works in Splink 4.0.0:
```
sql = f"""
with ranked as

(                                             
select *,                                                    
row_number() OVER ( 
    PARTITION BY simulant_id_l order by match_weight desc 
    ) as row_number 
from {results.physical_name} 
) 

select * 
from ranked 
where row_number = 1
"""
results = linker.query_sql(sql)  
```               

The error message is:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'Linker' object has no attribute 'query_sql'
```

I am using Windows 10. The backend is the default DuckDB. Please advise.

**Comment by RobinL** [comment, marked as answer]
Thanks for spotting - in the new api it's `linker.misc.query_sql(sql)`, but we'd forgotten to update the docs.

Fixed in #2322
